src/components/common/AddressInputReadOnly/index.tsx

```diff
@@ -0,0 +1,44 @@
+import { useState, type ReactElement } from 'react'
+import { IconButton, InputAdornment, InputLabel, OutlinedInput, SvgIcon, Typography } from '@mui/material'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import SaveAddressIcon from '@/public/images/common/save-address.svg'
+import css from './styles.module.css'
+import EntryDialog from '@/components/address-book/EntryDialog'
+import useAddressBook from '@/hooks/useAddressBook'
+
+const AddressInputReadOnly = ({ label, address }: { label: string; address: string }): ReactElement => {
```

Comment:
A lot of this code is similar to the `AddressBookInput` where we have a read only flag. Can we use that instead?

Reply:
They seem too different to me to abstract them. While `AddressBookInput` contains the Autocomplete component and deals mostly with the shown options, the `AddressInputReadOnly` is more dumb and it only displays the Address in the redesign style.